server/src/main/java/io/crate/execution/ddl/SwapRelationsOperation.java

```diff
@@ -68,25 +69,28 @@ public UpdatedState execute(ClusterState state, SwapRelationsRequest swapRelatio
         if (dropRelations.isEmpty()) {
             return stateAfterRename;
         } else {
-            return applyDropRelations(state, stateAfterRename, dropRelations);
+            return applyDropRelations(stateAfterRename, dropRelations);
         }
     }
 
-    private UpdatedState applyDropRelations(ClusterState state, UpdatedState updatedState, List<RelationName> dropRelations) {
+    private UpdatedState applyDropRelations(UpdatedState updatedState, List<RelationName> dropRelations) {
         ClusterState stateAfterRename = updatedState.newState;
         Metadata.Builder updatedMetadata = Metadata.builder(stateAfterRename.metadata());
         RoutingTable.Builder routingBuilder = RoutingTable.builder(stateAfterRename.routingTable());
 
-        Metadata metadata = state.metadata();
         for (RelationName dropRelation : dropRelations) {
             for (Index index : IndexNameExpressionResolver.concreteIndices(
-                    metadata, IndicesOptions.LENIENT_EXPAND_OPEN, dropRelation.indexNameOrAlias())) {
+                    stateAfterRename.metadata(), IndicesOptions.LENIENT_EXPAND_OPEN, dropRelation.indexNameOrAlias())) {
```

Comment:
if I got it correctly, this is addressing `non_partitioned_source` case.

why this affected/fixed case `test_swap_source_nonpartitioned_target_partitioned_with_drop_source`

but existing before this PR `test_swap_two_simple_tables_with_drop_source` was not affected by this change?

Reply:
Check the diff for simple tables:
![simple_tables](https://user-images.githubusercontent.com/5058131/209330008-71424d4e-dd66-4283-b0e7-f33a2fffc1e3.png)
vs parted source <-> simple target:
![partedsource_simpletarget](https://user-images.githubusercontent.com/5058131/209330127-7820a741-b060-4732-841c-11d443deda11.png)

For simple cases, you can still find both source and target in original cluster state, and updated cluster state.

Reply:
For the parted source <-> simple target
We added the `source` partitions (found in original state) to the new indices, and therefore they were not dropped

For the simple source <-> parted target
source becomes parted and needs to be dropped, but we didn't drop the template.

Reply:
I see, thx for the clarification!